packages/components/src/toggle-control/index.tsx

```diff
@@ -64,8 +64,19 @@ export function ToggleControl( {
 
 	let describedBy, helpLabel;
 	if ( help ) {
-		describedBy = id + '__help';
-		helpLabel = typeof help === 'function' ? help( checked ) : help;
+		if ( typeof help === 'function' ) {
+			// `help` as a function works only for controlled components where
+			// `checked` is passed down from parent component. Uncontrolled
+			// component can show only a static help label.
+			if ( checked !== undefined ) {
+				helpLabel = help( checked );
```

Comment:
I wonder if it would make sense to allow `help` to be called with an `undefined`, as it currently is on `trunk`. The docs don't mention anything about `help` working only for controlled components

Reply:
> could you add a CHANGELOG entry too?

Added an entry about the `ToggleControl`'s `help` prop type. Other changes in this PR are not publicly visible.

Reply:
> I wonder if it would make sense to allow `help` to be called with an `undefined`

That's an incorrect behavior though. If the `checked` prop is not specified, then the `input` in uncontrolled, i.e., it stores the `checked` state internally and only fires the `onChange` events. The user can check the toggle, it will fire the `onChange` event, will be checked visually, but the `change` variable will still be `undefined`. The wrong `help` will be shown.

After a closer look, I see that `ToggleControl` doesn't really support uncontrolled mode correctly. A checked `FormToggle` component won't have the `is-checked` CSS class, because the `checked` prop is never `true`. Also the component doesn't have a `defaultChecked` prop for setting initial uncontrolled value.

The `checked` value shouldn't be allowed to be `undefined`. It only creates an illusion of supporting uncontrolled mode, but the support is not really there. And all usages in the Gutenberg codebase seem to be controlled.

Reply:
Thank you again for looking into this! Your reasoning makes sense to me.

> After a closer look, I see that ToggleControl doesn't really support uncontrolled mode correctly.

That seems to be an issue most of the older control components in the library, thank you for flagging